_shared/local_extensions.py

```diff
@@ -158,6 +160,57 @@ def dedent(self, s):
         """Return `s` dedented."""
         return dedent(s)
 
+    def random_port_number(self):
+        return random.choice(
+            [
+                i
+                for i in range(1025, 10000)  # Ports <=1024 are privileged on Linux.
+                if i
+                not in (
+                    # Commonly-used port numbers to avoid.
+                    8080,  # Common alternative HTTP port.
+                    # Port numbers already used by existing Hypothesis apps.
+                    5000,  # h's web process.
```

Comment:
The code formatting is Black's, not mine :man_shrugging: Including the lack of empty lines between the different sections of the list.